utils/filesystem.py

```diff
@@ -92,9 +93,16 @@ def remove_directory_if_empty(path):
 
 
 def create_directories(path):
-    if not os.path.exists(path):
-        try:
-            os.makedirs(path)
-        except OSError as exc:
-            if exc.errno != errno.EEXIST:
-                raise
+    try:
+        os.makedirs(path)
+    except OSError as exc:
+        if exc.errno != errno.EEXIST:
+            raise  # Ignore exception if directory already existing
+
+
+def remove_directory(path):
```

Comment:
Perhaps we can make this name a bit clearer - it's not clear that it will recursively remove a directory, or ignore errors.
Can we add simple docstrings to the two directory methods?